scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala

```diff
@@ -101,14 +105,34 @@ object Scalafmt {
       file: String,
       range: Set[Range] = Set.empty
   ): Try[String] =
-    if (!FileOps.isAmmonite(file)) doFormatOne(code, style, file, range)
-    else {
+    if (FileOps.isAmmonite(file)) {
       // XXX: we won't support ranges as we don't keep track of lines
       val chunks = ammonitePattern.split(code)
       if (chunks.length <= 1) doFormatOne(code, style, file, range)
       else
         flatMapAll(chunks.iterator)(doFormatOne(_, style, file))
           .map(_.mkString("\n@\n"))
+    } else if (FileOps.isMarkdown(file)) {
+      if (code.contains("```scala mdoc")) {
```

Comment:
This is a small performance-related piece that Olafurpg sugested (If I understood him correctly), to bypass parsing and rendering markdown files that don't have any affected fenced blocks.

Reply:
Thanks, done.